rust/otap-dataflow/crates/engine/src/message.rs

```diff
@@ -302,7 +302,7 @@ impl<PData> MessageChannel<PData> {
                         // Begin draining mode, but don’t return Shutdown yet
                         let when = Instant::now() + deadline;
                         self.shutting_down_deadline = Some(when);
-                        self.pending_shutdown = Some(ControlMsg::Shutdown { deadline: Duration::ZERO, reason });
+                        self.pending_shutdown = Some(ControlMsg::Shutdown { deadline, reason });
```

Comment:
@lquerel I'm not sure if this was a bug, but I noticed that this was causing the `deadline` that gets passed into the shutdown handler to always be 0, even if in our tests we do something like:

```rs
                test_context.send_shutdown(
                  Duration::from_secs(1),  // something non-zero
                  "test completed"
                  )
                    .await
                    .unwrap();
```
Can you confirm if this is the right fix for this?

Reply:
Yes, I think so. In any case, there’s a good chance that I will revisit this part of the code.